src/mscorlib/src/System/Resources/ResourceSet.cs

```diff
@@ -182,15 +182,17 @@ public virtual Type GetDefaultReader()
             return typeof(ResourceReader);
         }
     
-#if !FEATURE_CORECLR
         // Returns the preferred IResourceWriter class for this kind of ResourceSet.
         // Subclasses of ResourceSet using their own Readers &; should override
         // GetDefaultReader and GetDefaultWriter.
         public virtual Type GetDefaultWriter()
         {
+#if FEATURE_CORECLR
+            return Type.GetType("System.Resources.ResourceWriter, System.Resources.Writer, Version=4.0.1.0, Culture=neutral, PublicKeyToken=" + AssemblyRef.MicrosoftPublicKeyToken, throwOnError: true);
```

Comment:
Do we also need to expose `IResourceWriter` interface from here for the ResourceWriter to implement? Note that there is code in Reflection.Emit that calls IResourceWriter.

I would be useful to physically exclude ResourceWriter from the CoreCLR build to make sure that it is not leaking anywhere.